### PR TITLE
[gif-search] Add Copying GIF File to Clipboard and support for The Finer Gifs Club

### PR DESCRIPTION
**Pull request by josephschmitt**
## Description

1. Adds support for copying the GIF file itself (instead of the URL) to the clipboard via AppleScript.
1. Adds a "Secondary Action" preference to change the action on CMD + ENTER
1. Adds a new command to search for GIFs from [The Finer Gifs Club](https://thefinergifs.club). This one's especially meaningful to me since I [created The Finer Gifs Club](https://joe.sh/in-search-of-finer-gifs) a few years ago to try and be the definitive search engine for finding gifs from The Office. It works just like searching for gifs from giphy.com and tenor.com, it's just limited to gifs from the show The Office. <br><br>Adding support was pretty standard, but had to make a couple of changes in order to accommodate not having a few fields in the `IGif` interface.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

File: extensions/gif-search/src/models/gif.ts [hunk 1]
```diff
@@ -4,15 +4,16 @@ import type { IGif as GiphyGif } from "@giphy/js-types";
 import { environment } from "@raycast/api";
 
 import type { TenorGif } from "./tenor";
+import { FinerGif } from "./finerthingsclub";
 
 export interface IGif {
   id: string | number;
   title: string;
-  url: string;
+  url?: string;
```
Comment (josephschmitt) [inline extensions/gif-search/src/models/gif.ts:12]: The Finer Gifs club has no standalone page for a single gif, so make this field optional

File: extensions/gif-search/src/models/gif.ts [hunk 2]
```diff
@@ -4,15 +4,16 @@ import type { IGif as GiphyGif } from "@giphy/js-types";
...
   slug: string;
   preview_gif_url: string;
   gif_url: string;
-  attribution:
+  attribution?:
```
Comment (josephschmitt) [inline extensions/gif-search/src/models/gif.ts:16]: No attribution needed for the finer gifs club, so optional here, too

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @josephschmitt 👋 

Thanks for the contribution 💪 I have some feedback and suggestions for you to address:

- package.json
  - Remove the `tenor-apiKey` since it's not in use.
  
- tenor.tsx 
  - `isShowingDetail` - Could be an idea to check if showPreview and if there is any items instead of isLoading, it makes a bad UX
 
The same goes for the two other 💪 

Looking forward to testing it out again 💯 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by josephschmitt** [comment]
> package.json
> 
> * Remove the `tenor-apiKey` since it's not in use.

Sorry not sure what you mean. I'm using this preference when someone uses the Tenor Search command.

**Comment by josephschmitt** [comment]
@pernielsentikaer 
> tenor.tsx
> 
> * `isShowingDetail` - Could be an idea to check if showPreview and if there is any items instead of isLoading, it makes a bad UX

Done! Good catch!

**Comment by josephschmitt** [comment]
Just bumping this PR from last week in case it got lost in your inboxes... @pernielsentikaer @fe9lix. Thanks!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @josephschmitt 👋 

Thanks for your patience and corporation trying things out, it's still some pretty new features that might need some tweaking, thanks for helping out 💪 

I have some feedback from the spin I gave it ready for you 😄 I also wrote you in [Slack](https://raycastcommunity.slack.com/) if you want to discuss anything at all.

- For all of them
  - After testing it out, I think all the <List> elements should have `isShowingDetail={true}` instead, I couldn't find anything in the documentation saying that it's illegal to do _(I'll get back to you with this one)_. If this is not set, then there is a lot of flicker while searching that we can avoid. 
  
- The Finer GIFs Club
  -  When deleting the search query, it should then go back to “No Results”

- Giphy
  - The preference `API Key` should be type password instead of text

- Tenor
  - The preference `API Key` should be type password instead of text

We're now adding support for [categories](https://developers.raycast.com/basics/prepare-an-extension-for-store#categories), [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) and [changelog](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) – do you have time for adding this too, I can do the screenshots if you want help 😄 

---
Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by josephschmitt** [comment]
> * After testing it out, I think all the  elements should have `isShowingDetail={true}` instead, I couldn't find anything in the documentation saying that it's illegal to do _(I'll get back to you with this one)_. If this is not set, then there is a lot of flicker while searching that we can avoid.

Had to keep the check for the results otherwise we don't get the correct "No Results" screen, so mostly just gets rid of the `isLoading` check.

**Review by pernielsentikaer — APPROVED** [review]
Hi @josephschmitt 👋 

Thanks for the update, much better 🔥 

Tested it all, everything works as expected, awesome 💪

**Pull request merged**